openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,191 @@
+        }
+        icc_user = {}
+
+        try:
+            user, _ = get_user(email=data.get("user_email"))
+            request_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_CREATE_USER_API_FUNCTION,
+                "users[0][username]": user.username + self._get_random_string(
+                    self.USERNAME_SUFFIX_LENGTH) if duplicated_username else user.username,
+                "users[0][password]": self._get_random_string(settings.ICC_HASH_LENGTH),
+                "users[0][firstname]": user.first_name,
+                "users[0][lastname]": user.last_name,
+                "users[0][email]": user.email,
+                "users[0][auth]": "saml2",
+            }
+
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=request_data,
+            )
+            log_details["request_payload"] = request_data
+        except Exception as error:  # pylint: disable=broad-except
+            LOG.error("Failed to create ICC user. Reason: %s", str(error))
+            log_details["request_payload"]["wstoken"] = "icc-api-token"
+            log_details["response"] = {"error": "Failed to create ICC user. Reason: " + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, "create_user")
+
+        return icc_user
+
+    def _validate_icc_user(self, icc_user):
+        """Ensure the user creation avoiding failure for username duplicates"""
+        duplicated_username = False
+        call_counter = 0
+        while not icc_user and call_counter <= self.USER_CREATION_MAX_RETRIES:
+            icc_user = self._create_icc_user(data, duplicated_username)
+            duplicated_username = True
+            call_counter += 1
+
+        return icc_user
+
+    def _get_icc_user_from_xml_response(self, response, method_type):
+        """
+        Method that receives a xml response and convert it to json object and returns it.
+        """
+        icc_user = {}
+        OBJECT_POSITION = 0
+
+        try:
+            str_response = xmltodict.parse(response.content)
+            json_str_response = json.dumps(str_response)
+            json_obj_response = json.loads(json_str_response)
+
+            if method_type == "create_user":
+                fields = json_obj_response["RESPONSE"]["MULTIPLE"]["SINGLE"]["KEY"]
+            else:
+                fields = json_obj_response["RESPONSE"]["SINGLE"]["KEY"][OBJECT_POSITION]["MULTIPLE"]["SINGLE"]["KEY"]
+
+            for field in fields:
+                if field["@name"] == "id":
+                    icc_user['id'] = field["VALUE"]
+                if field["@name"] == "username":
+                    icc_user["username"] = field["VALUE"]
```

Comment:
Add a .get method when you are trying to get the value of a dict. @ivanvgh
Also, this can be an inline "if" assignment.

Reply:
Done.